src/analysis/retail/monk/mistweaver/normalizers/CastLinkNormalizer.ts

```diff
@@ -75,20 +75,28 @@ const EVENT_LINKS: EventLink[] = [
     referencedEventId: SPELLS.RENEWING_MIST_HEAL.id,
     referencedEventType: [EventType.ApplyBuff, EventType.RefreshBuff],
     backwardBufferMs: MAX_REM_DURATION,
+    additionalCondition(linkingEvent, referencedEvent) {
+      return (
+        (linkingEvent as RemoveBuffEvent).targetID ===
+          (referencedEvent as ApplyBuffEvent).targetID ||
+        (linkingEvent as RemoveBuffEvent).targetID ===
+          (referencedEvent as RefreshBuffEvent).targetID
+      );
+    },
   },
-  // link ReM to an EnvM/RSK cast
+  // link ReM application from Rapid diffusion
   {
     linkRelation: FROM_RAPID_DIFFUSION,
     linkingEventId: [SPELLS.RENEWING_MIST_HEAL.id],
     linkingEventType: [EventType.ApplyBuff, EventType.RefreshBuff],
     referencedEventId: [
-      TALENTS_MONK.RISING_SUN_KICK_TALENT.id,
       TALENTS_MONK.ENVELOPING_MIST_TALENT.id,
-      SPELLS.ENVELOPING_MIST_TFT.id,
+      TALENTS_MONK.RISING_SUN_KICK_TALENT.id,
     ],
     referencedEventType: [EventType.Cast],
-    backwardBufferMs: CAST_BUFFER_MS,
+    backwardBufferMs: 300,
```

Comment:
Arbitrary number that I assume is from experimentation. Let's either add a comment or make the 300 into a global variable with name `MAXIMUM_RAPID_DIFFUSION_BUFFER` or something similar